packages/edit-site/src/components/routes/link.js

```diff
@@ -32,12 +32,20 @@ export default function Link( {
 	state,
 	replace: shouldReplace = false,
 	children,
+	onClick: onClickProp,
 	...props
 } ) {
 	const { href, onClick } = useLink( params, state, shouldReplace );
 
 	return (
-		<a href={ href } onClick={ onClick } { ...props }>
+		<a
+			href={ href }
+			onClick={ ( event ) => {
+				onClickProp( event );
```

Comment:
The only change here is the addition of the onClick prop in case we want to react to link clicks to perform other related actions.